### ISSUE TITLE
AtomicPut doesn't compare value

### ISSUE THREAD

**Issue by mthenw** [abronan/libkv#25]
Right now AtomicPut doesn't compare value, just ModRevision. Is it intended or not implemented yet? :)

**Comment by abronan** [comment]
You're right, probably just an oversight, vacation didn't help 🏖 😅. I'll look into the remaining issues when I'm back.

**Comment by mgood** [comment]
Could you elaborate on what you would want to gain by comparing the value as well?

I'm more familiar with Consul, and there it is intentional to only compare the index. The Consul CAS parameter only takes an index since modifying the value would also modify the index. AFAICT comparing the value would only provide a different result if you passed in an index/value combination that wasn't a valid state that existed in the store. Otherwise it seems like comparing the value would be redundant and slow down operations.
